GitUI/BranchTreePanel/RepoObjectsTree.Nodes.Remotes.cs

```diff
@@ -26,19 +28,24 @@ private void UiCommands_GitUICommandsChanged(object sender, GitUICommandsChanged
                 TreeViewNode.TreeView.SelectedNode = null;
             }
 
-            protected override void LoadNodes(CancellationToken token)
+            protected override async Task LoadNodesAsync(CancellationToken token)
             {
+                await TaskScheduler.Default;
+                token.ThrowIfCancellationRequested();
                 var nodes = new Dictionary<string, BaseBranchNode>();
 
                 var branches = Module.GetRefs()
                     .Where(branch => branch.IsRemote && !branch.IsTag)
                     .OrderBy(r => r.Name)
                     .Select(branch => branch.Name);
 
+                token.ThrowIfCancellationRequested();
                 var remotes = Module.GetRemotes(allowEmpty: true);
+                token.ThrowIfCancellationRequested();
```

Comment:
I paused to consider the seemingly high number of checks here and concluded that I should put more in my own code.

They're harmless, quick to execute and help keep the UI responsive.

42 seems useful (it divides two long-running activities), but 44 not so much as 48 will pick it up.

Reply:
I made the same observations about my code...
However we need to be mindful not to turn the codebase into `token.ThrowIfCancellationRequested();` fest as it may inhibit overall readability